src/gsy_e/models/area/scm_manager.py

```diff
@@ -0,0 +1,249 @@
+import logging
+from dataclasses import dataclass
+from typing import Dict
+from uuid import uuid4
+
+from gsy_framework.data_classes import Trade
+from pendulum import DateTime
+
+from gsy_e.constants import DEFAULT_GRID_SELLER_STRING, DEFAULT_SCM_SELLER_STRING
+
+
+@dataclass
+class HomeAfterMeterData:
+    # pylint: disable=too-many-instance-attributes
+    """Represent all the after meter data for a single home area."""
+    home_uuid: str
+    home_name: str
+    sharing_coefficient_percent: float = 0.
+    grid_fees_eur: float = 0.
+    market_maker_rate_eur: float = 0.
+    feed_in_tariff_eur: float = 0.
+    consumption_kWh: float = 0.
+    production_kWh: float = 0.
+    self_consumed_energy_kWh: float = 0.
+    energy_surplus_kWh: float = 0.
+    energy_need_kWh: float = 0.
+    community_production_kWh: float = 0.
+    trades = None
+
+    def __post_init__(self):
+        self.self_consumed_energy_kWh = min(self.consumption_kWh, self.production_kWh)
```

Comment:
Can you please explain this step? Why would the production influence the consumed energy?

Reply:
Heads up on the double self :D The member variable is named `self_consumed_energy_kWh` , which reflects the amount of energy that the house would self-consume (in other words, consumption that would be covered by production of the house without external trading). Self consumption can never be greater than the production of the house, if the house produces more than it can consume it will sell the remaining energy to the grid, and the self_consumed_energy_kWh will be equal to the consumption_kWh

Reply:
Exactly!